CMakeLists.txt

```diff
@@ -232,6 +243,85 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(core_cxx_flags)
+# Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
+try_append_cxx_flags(core_cxx_flags "-fno-extended-identifiers")
+
+# Currently all versions of gcc are subject to a class of bugs, see the
+# gccbug_90348 test case (only reproduces on GCC 11 and earlier) and
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111843. To work around that, set
+# -fstack-reuse=none for all gcc builds. (Only gcc understands this flag).
+try_append_cxx_flags(core_cxx_flags "-fstack-reuse=none")
+
+separate_arguments(core_cxx_flags)
+target_compile_options(core INTERFACE ${core_cxx_flags})
+unset(core_cxx_flags)
+
+if(HARDENING)
+  if(MSVC)
+    try_append_linker_flag(core "/DYNAMICBASE")
+    try_append_linker_flag(core "/HIGHENTROPYVA")
+    try_append_linker_flag(core "/NXCOMPAT")
+  else()
+    set(hardened_cxx_flags)
+    try_append_cxx_flags(hardened_cxx_flags "-Wstack-protector")
+    try_append_cxx_flags(hardened_cxx_flags "-fstack-protector-all")
+    try_append_cxx_flags(hardened_cxx_flags "-fcf-protection=full")
+    separate_arguments(hardened_cxx_flags)
+    target_compile_options(core INTERFACE ${hardened_cxx_flags})
+
+    if(MINGW)
+      target_link_libraries(core INTERFACE ssp)
+    else()
+      # stack-clash-protection doesn't currently work, and likely should just be skipped for Windows.
+      # See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90458 for more details.
+      try_append_cxx_flags(hardened_cxx_flags "-fstack-clash-protection")
+    endif()
+
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+      try_append_cxx_flags(hardened_cxx_flags "-mbranch-protection=bti")
```

Comment:
Is this working? I don't see it in "Common compile options", or any other output, on two different aarch64 machines.